File: 2200-find-all-k-distant-indices-in-an-array/2200-find-all-k-distant-indices-in-an-array.cpp
class Solution {
public:
    vector<int> findKDistantIndices(vector<int>& nums, int key, int k) {
        vector<int> ans;
        queue<int> q;
        for(int j=0;j<nums.size();j++) {
            if(nums[j]==key)
                q.push(j);
        }
        for(int i=0;i<nums.size() && q.size()>0;i++) {
         //    if(q.empty()) break;
            int j = q.front();
            if(abs(i-j)<=k) 
                ans.push_back(i);
            else {
                if(i>j) {
                    q.pop();
                
                //start scanning from current index 
                    i--;
                }
            }
        }
        return ans;
    }
};

File: 2200-find-all-k-distant-indices-in-an-array/2200-find-all-k-distant-indices-in-an-array.java
class Solution {
    public List<Integer> findKDistantIndices(int[] nums, int key, int k) {
        List<Integer> jindex = new ArrayList<>();
        for(int i=0;i<nums.length;i++) {
            if(key==nums[i]);
            jindex.add(i);
        }
        Set<Integer> set = new HashSet<>();
        for(int i=0;i<nums.length;i++) {
            for(int j: jindex) {
                if(Math.abs(i-j)<=k && nums[j]==key) {
                    set.add(i);
                }
            }
        }
        List<Integer> result = new ArrayList<>(set);
        Collections.sort(result);
        return result;
    }
}